sdk/storage/blobs/inc/az_storage_blobs.h

```diff
@@ -0,0 +1,147 @@
+AZ_NODISCARD AZ_INLINE az_span az_storage_blobs_blob_header_Content_Length() { return AZ_STR("Content_Length"); }
+
+AZ_NODISCARD AZ_INLINE az_span az_storage_blobs_blob_type_AppendBlob() { return AZ_STR("AppendBlob");} /* UnSupported */
+AZ_NODISCARD AZ_INLINE az_span az_storage_blobs_blob_type_BlockBlob() { return AZ_STR("BlockBlob");}
+AZ_NODISCARD AZ_INLINE az_span az_storage_blobs_blob_type_PageBlob() { return AZ_STR("PageBlob"); } /* UnSupported */
+
+typedef struct {
+  az_span service_version;
+  az_http_policy_retry_options retry;
+} az_storage_blobs_blob_client_options;
+
+typedef struct {
+  az_span option;
+} az_storage_blobs_blob_upload_options;
+
+typedef struct {
+  az_span option;
+} az_storage_blobs_blob_download_options;
+
+typedef struct {
+  az_span uri;
+  az_span blob_type;
+  az_http_pipeline pipeline;
+  az_storage_blobs_blob_client_options client_options;
+} az_storage_blobs_blob_client;
+
+extern az_storage_blobs_blob_client_options const AZ_STORAGE_BLOBS_BLOB_CLIENT_DEFAULT_OPTIONS;
+
+/**
+ * @brief Init a client with default options
+ * This is convinient method to create a client with basic settings
+ * Options can be updated specifally after this for unique customization
+ *
+ * Use this, for instance, when only caring about setting one option by calling this method and then
+ * overriding that specific option
+ */
+AZ_NODISCARD AZ_INLINE az_result
+az_storage_blobs_blob_client_options_init(az_storage_blobs_blob_client_options * const options) {
+  AZ_CONTRACT_ARG_NOT_NULL(options);
+  *options = AZ_STORAGE_BLOBS_BLOB_CLIENT_DEFAULT_OPTIONS;
+  return AZ_OK;
+}
+
+AZ_NODISCARD AZ_INLINE az_result az_storage_blobs_blob_client_init(
+    az_storage_blobs_blob_client * const client,
+    az_span const uri,
+    void * const credential,
+    az_storage_blobs_blob_client_options const * const options) {
+  AZ_CONTRACT_ARG_NOT_NULL(client);
+
+  client->uri = uri;
+  // use default options if options is null. Or use customer provided one
+  if (options == NULL) {
+    client->client_options = AZ_STORAGE_BLOBS_BLOB_CLIENT_DEFAULT_OPTIONS;
+  } else {
+    client->client_options = *options;
+  }
+
+  client->pipeline = (az_http_pipeline){
+    .policies = {
+      { .pfnc_process = az_http_pipeline_policy_uniquerequestid, .data = NULL },
+      { .pfnc_process = az_http_pipeline_policy_retry, .data = &client->client_options.retry },
+      { .pfnc_process = az_http_pipeline_policy_authentication, .data = credential },
+      { .pfnc_process = az_http_pipeline_policy_logging, .data = NULL },
+      { .pfnc_process = az_http_pipeline_policy_bufferresponse, .data = NULL },
+      { .pfnc_process = az_http_pipeline_policy_distributedtracing, .data = NULL },
+      { .pfnc_process = az_http_pipeline_policy_transport, .data = NULL },
+      { .pfnc_process = NULL, .data = NULL },
+    }, 
+    };
+
+  // Currently only BlockBlobs are supported
+  client->blob_type = az_storage_blobs_blob_type_BlockBlob();
```

Comment:
I don't think you should need this field.
A specific operation might need the blob type but this is on the operation and not part of the client

Reply:
We will need it if we ever support additional blob types in the future

Reply:
We don't have this in any other language. So, we might need it if we have a very different design but not if we are attempting consistency. We can discuss on phone.